bridge_adaptivity/api/backends/openedx.py

```diff
@@ -117,71 +119,104 @@ def get_provider_courses(self, username=None, org=None, mobile=None):
             org=org,
             mobile=mobile
         )
-        return resource.get('results')
+        return filter_function(resource.get('results'))
 
 
-def get_available_blocks(course_id):
+def check_source_id(source_id, not_allow_empty_source_id=True):
+    """
+    Check that passed source_id parameter is valid.
+
+    If there's only one active source provider - source_id parameter is not required, it will get first active.
+    :param source_id: LtiConsumer object id
+    :return: queryset of content_sources
+    :raise HttpClientError: if provided parameters are not valid
+    """
+    content_sources = get_content_providers(source_id=source_id)
+    if not content_sources:
+        # if no active content sources
+        raise HttpClientError(_("No active Content Provider"))
+
+    if not_allow_empty_source_id:
```

Comment:
I would move check for mandatory param to the top of the function to avoid odd database requests if the condition is True